packages/api/src/routes/medical/internal-cq.ts

```diff
@@ -10,35 +11,35 @@ import { Config } from "../../shared/config";
 import { asyncHandler } from "../util";
 
 const apiKey = Config.getCQApiKey();
-const apiMode = Config.getEnvType();
 
 dayjs.extend(duration);
 
 const router = Router();
 
 /**
- * GET /internal/carequality
+ * POST /internal/carequality/rebuild-directory
  *
  * Retrieves organizations from the Carequality Directory and uploads them into our database.
  * @returns Returns the number of organizations fetched, how many are newly-added and how many updated.
  */
-router.get(
-  "/",
+router.post(
+  "/rebuild-directory",
   asyncHandler(async (req: Request, res: Response) => {
-    const cq = new Carequality(apiKey, apiMode);
+    const cq = new Carequality(apiKey);
     const resp = await cq.listAllOrganizations();
-    const orgs = parseCQDirectoryEntries(resp.organizations);
+    const orgs = parseCQDirectoryEntries(resp);
 
     const response = {
-      totalFetched: resp.count,
+      totalFetched: resp.length,
       added: 0,
       updated: 0,
     };
 
-    for (const org of orgs) {
+    const directoryEntryPromises = executeAsynchronously(orgs, async org => {
       const dbResponse = await createOrUpdateCQDirectoryEntry(org);
       dbResponse.updated ? response.updated++ : response.added++;
-    }
+    });
```

Comment:
How many orgs are we expecting to get from `listAllOrganizations()`? Are we ready to process them all in parallel against our DB?

I mentioned "limiting the amount of promises we run at the same time". If we're sure `orgs` can have at most a couple handful records, this is fine, otherwise we need to determine/limit the max amount of promises we can run in parallel here.

Reply:
Ohhh I see. Yea I think we expect up to 40k entries. 

Need to use `numberOfParallelExecutions`, I assume. How do I decide how many to run simultaneously? i see some places using just 10.

Reply:
Yeah, it all depends on what we on each Promise. In this case:
- we're querying and updating the DB;
- AFAIK, we expect to have only one request for this endpoint being executed at any given time;
...so I think we can add some load there, b/c its only one HTTP request at a time, but still control so we don't (1) consume all our DB connections from the DB connection pool and (2) don't overload the Node's event loop with 40K promises at the same time.

Maybe start with 20 and see how it performs? It should be simple to update this later, likely a const at the top of the file.

If we have a place on Notion to store references to how to go about integrating to CQ, we could add this there on a section about running/scaling CQ integration: "consider the amount of parallel processing of CQ directory entries on endpoint XYZ".